plugins/whoami.yaml

```diff
@@ -0,0 +1,41 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: whoami
+spec:
+  version: "v0.0.2"
+  platforms:
+  - selector:
+      matchLabels:
+        os: darwin
+        arch: amd64
+    uri: https://github.com/rajatjindal/kubectl-whoami/releases/download/v0.0.2/darwin-amd64-v0.0.2.tar.gz
+    sha256: 5eb338fd2c97b4cd54a1f1feddc66ef6696e49f11b811e3e8229622a474cc4e6
+    files:
+    - from: "*"
+      to: "."
+    bin: kubectl-whoami
+  - selector:
+      matchLabels:
+        os: linux
+        arch: amd64
+    uri: https://github.com/rajatjindal/kubectl-whoami/releases/download/v0.0.2/linux-amd64-v0.0.2.tar.gz
+    sha256: a83afa5def7762062fcb301de3d31a8e6542330217159583f7dabf7acea417a7
+    files:
+    - from: "*"
+      to: "."
+    bin: kubectl-whoami
+  shortDescription: find subject of effective kubeconfig context
```

Comment:
Also FWIW there are ways to modify the current authenticated user that doesn't come from kubeconfig.

e.g. `--token` option to kubectl, or setting env vars like GOOGLE_APPLICATION_CREDENTIALS would override the current user.

Reply:
but that's a philosophical problem. I don't know how you'd address it.

You can just simply say:

> Show the subject that's currently authenticated as.

Reply:
it does cover --token scenario

the GOOGLE_APPLICATION_CREDENTIALS is done via the external auth-provider or via the client-go?

I think I should check if basicauth or certauth is enabled, and in other scenario just call tokenreview request using client-go and let that sort out the auth provider workflow.

i will try it out with a test cluster and other auth providers before merging. thank you for your review.

Reply:
My point: —token is not in kubeconfig. 
But the description says it uses kubeconfig.

Reply:
got it. thank you

Reply:
updated the description. I still haven't had a chance to test other providers. Will be working on that as soon as I've some more time in hand.